Libraries/libutil/Tests/test_MemoryFilesystem.cpp

```diff
@@ -251,21 +207,109 @@ TEST(MemoryFilesystem, Write)
     EXPECT_FALSE(filesystem.exists("/invalid/new"));
 }
 
-TEST(MemoryFilesystem, ResolvePath)
+TEST(MemoryFilesystem, CopyFile)
 {
+    std::vector<uint8_t> contents;
     auto filesystem = BasicFilesystem();
-    EXPECT_EQ(filesystem.resolvePath("/"), "/");
-    EXPECT_EQ(filesystem.resolvePath("//"), "/");
-    EXPECT_EQ(filesystem.resolvePath("/./"), "/");
-    EXPECT_EQ(filesystem.resolvePath("//file1"), "/file1");
-    EXPECT_EQ(filesystem.resolvePath("/dir1"), "/dir1");
-    EXPECT_EQ(filesystem.resolvePath("/dir1/"), "/dir1/");
-    EXPECT_EQ(filesystem.resolvePath("/dir1/.."), "/");
-    EXPECT_EQ(filesystem.resolvePath("/dir1//file2"), "/dir1/file2");
-    EXPECT_EQ(filesystem.resolvePath("/dir2/.././dir1/file2"), "/dir1/file2");
+
+    /* Must copy to a real path. */
+    EXPECT_FALSE(filesystem.copyFile("/file1", "/invalid/file1"));
+
+    /* Can't copy over a directory, even an empty one. */
+    EXPECT_FALSE(filesystem.copyFile("/file1", "/dir1"));
+    EXPECT_FALSE(filesystem.copyFile("/file1", "/dir2/dir3"));
+
+    /* Can copy to a new path. */
+    contents.clear();
+    EXPECT_TRUE(filesystem.copyFile("/file1", "/copied"));
+    EXPECT_TRUE(filesystem.read(&contents, "/copied"));
+    EXPECT_EQ(contents, Contents("one"));
+
+    /* Can copy over an existing file. */
+    contents.clear();
+    EXPECT_TRUE(filesystem.copyFile("/file1", "/dir1/file2"));
+    EXPECT_TRUE(filesystem.read(&contents, "/dir1/file2"));
+    EXPECT_EQ(contents, Contents("one"));
+}
+
+TEST(MemoryFilesystem, RemoveFile)
+{
+    auto filesystem = BasicFilesystem();
+
+    EXPECT_TRUE(filesystem.exists("/file1"));
+    EXPECT_TRUE(filesystem.removeFile("/file1"));
+    EXPECT_FALSE(filesystem.exists("/file1"));
+
+    EXPECT_TRUE(filesystem.exists("/dir1/file2"));
+    EXPECT_TRUE(filesystem.removeFile("/dir1/file2"));
+    EXPECT_FALSE(filesystem.exists("/dir1/file2"));
+
+    EXPECT_FALSE(filesystem.removeFile("/dir1"));
```

Comment:
Probably ought to verify that if removeFile reports failure, the filesystem isn't accidentally mutated.